### PR TITLE
Coverity 1196421, 1196464, 1196465, 1215252: buffer_upload plugin

### PR DESCRIPTION
**Pull request by shukitchan**
CID 1196421: Unchecked return value from library
CID 1196464: Dereference before null check
CID 1196465: Dereference before null check
CID 1215252: Dereference null return value
for buffer_upload plugin

### CODE REVIEW

**Comment by shukitchan** [comment]
I forgot the clang-format. will fix it soon.

File: plugins/experimental/buffer_upload/buffer_upload.cc [hunk 1]
```diff
@@ -599,6 +598,9 @@ convert_url_func(TSMBuffer req_bufp, TSMLoc req_loc)
   }
 
   char *hostname = (char *)getenv("HOSTNAME");
```
Comment (jpeach) [inline plugins/experimental/buffer_upload/buffer_upload.cc]: `const char * hostname`

File: plugins/experimental/buffer_upload/buffer_upload.cc [hunk 2]
```diff
@@ -967,7 +969,10 @@ create_directory()
       goto error_out;
     }
     while ((d = readdir(dir))) {
-      remove(d->d_name);
+      if (remove(d->d_name) < 0) {
+        TSError("[buffer_upload] Unable to remove %s", d->d_name);
```
Comment (jpeach) [inline plugins/experimental/buffer_upload/buffer_upload.cc]:
```C
TSError("[buffer_upload] Unable to remove '%s': %s", d->d_name, strerror(errno));
```

Comment (jpeach) [inline plugins/experimental/buffer_upload/buffer_upload.cc]: Also, I assume that removing '.' and '..' will always fail here?

Comment (shukitchan) [inline plugins/experimental/buffer_upload/buffer_upload.cc]: yes. it should be.

**Comment by shukitchan** [comment]
@jpeach pls take a look again. will squash before merge

File: plugins/experimental/buffer_upload/buffer_upload.cc [hunk 3]
```diff
@@ -598,7 +597,10 @@ convert_url_func(TSMBuffer req_bufp, TSMLoc req_loc)
     return;
   }
 
-  char *hostname = (char *)getenv("HOSTNAME");
+  const char *hostname = (char *)getenv("HOSTNAME");
```
Comment (jpeach) [inline plugins/experimental/buffer_upload/buffer_upload.cc]:
```C
const char *hostname = getenv("HOSTNAME");
```

:)

**Comment by shukitchan** [comment]
[approve ci autest]

**Comment by zwoop** [comment]
Yeah, looks good to me, but please squash it down, and then we can approve it.

**Comment by shukitchan** [comment]
@jpeach any more things you want before i squash.

**Comment by shukitchan** [comment]
[approve ci autest]

**Comment by shukitchan** [comment]
@jpeach / @zwoop ... commits squashed!

**Pull request merged**